geolocation/ios/Plugin/GeolocationPlugin.swift

```diff
@@ -0,0 +1,200 @@
+        let removalQueue = callQueue.filter { $0.value == .permissions }
+        callQueue = callQueue.filter { $0.value != .permissions }
+
+        for (id, _) in removalQueue {
+            if let call = bridge?.getSavedCall(id) {
+                checkPermissions(call)
+                bridge?.releaseCall(call)
+            }
+        }
+    }
+
+    public func stopUpdating() {
+        self.locationManager.stopUpdatingLocation()
+    }
+
+    private func reportLocation(_ call: CAPPluginCall, _ locations: [CLLocation]) {
+        if let location = locations.first {
+            let result = makePosition(location)
+            call.resolve(result)
+        } else {
+            // TODO: Handle case where location is nil
+            call.resolve()
+        }
+    }
+
+    @objc func checkPermissions(_ call: CAPPluginCall) {
+        var status: String = ""
+
+        if CLLocationManager.locationServicesEnabled() {
+            switch CLLocationManager.authorizationStatus() {
+            case .notDetermined:
+                status = "prompt"
+            case .restricted, .denied:
+                status = "denied"
+            case .authorizedAlways, .authorizedWhenInUse:
+                status = "granted"
+            @unknown default:
+                status = "prompt"
+            }
+        } else {
+            call.reject("Location services are not enabled")
+            return
+        }
+
+        let result = [
+            "location": status
+        ]
+
+        call.resolve(result)
+    }
+
+    @objc func requestPermissions(_ call: CAPPluginCall) {
+        if CLLocationManager.locationServicesEnabled() {
+            // If state is not yet determined, request perms.
+            // Otherwise, report back the state right away
+            if CLLocationManager.authorizationStatus() == .notDetermined {
+                call.save()
+                callQueue[call.callbackId] = .permissions
+
+                DispatchQueue.main.async {
+                    self.locationManager.delegate = self
+                    self.locationManager.requestWhenInUseAuthorization()
+                }
+            } else {
+                checkPermissions(call)
+            }
+        } else {
+            call.reject("Location services are not enabled")
+        }
+    }
+
+    func makePosition(_ location: CLLocation) -> JSObject {
+        var ret = JSObject()
+        var coords = JSObject()
+        coords["latitude"] = location.coordinate.latitude
+        coords["longitude"] = location.coordinate.longitude
+        coords["accuracy"] = location.horizontalAccuracy
+        coords["altitude"] = location.altitude
+        coords["altitudeAccuracy"] = location.verticalAccuracy
+        coords["speed"] = location.speed
+        coords["heading"] = location.course
+        ret["timestamp"] = NSNumber(value: Int((location.timestamp.timeIntervalSince1970 * 1000)))
```

Comment:
The `Int` doesn't need to be wrapped in an NSNumber in Swift. It will be automatically converted for the JSON serialization.